### ISSUE TITLE
Errors happening in scripts created from blob URLs are not deduplicated

### ISSUE THREAD

**Issue by pornel** [getsentry/sentry#2430]
We load scripts from Blob URLs (`URL.createObjectURL(blob)`), because we need it as a work around AppCache bugs (and we're stuck with AppCache because of iOS). We're also moving to running code in WebWorkers soon, and Blob URLs are a good way to launch WebWorkers without additional requests.

However, Blob URLs make our scripts appear to originate from a unique random URLs such as `blob:http://app.ft.com/7f7aaadf-a006-4217-9ed5-5fbf8585c6c0`, where the UUID is different every time for every user.

This seems to defeat error grouping Sentry, and since we've made this change we're drowning in duplicate errors:

![screen shot 2015-12-09 at 12 57 47](https://cloud.githubusercontent.com/assets/72159/11685905/7d3a40b8-9e74-11e5-9471-e2143aa52739.png)

Could you normalize Blob URLs and ignore the random portion?

Is there an API in Raven that would allow us to provide you a mapping from Blob URL to an HTTP URL?

**Comment by dcramer** [comment]
Is this on getsentry.com? If so could you toss a link so we can look at the raw JSON? Alternatively, can you grab the JSON stacktrace component?

As an aside, you can use the 'fingerprint' attribute to override the hashing algorithm.

**Comment by dcramer** [comment]
I'm going to close this out as it's more a question and needs more information, but I will keep an eye out for updates.

Of note here, the blob url you're showing in this list as not used for hashing. We use stacktrace/message components, and the blob here is the 'context'

**Comment by pornel** [comment]
It is on getsentry.com.

```js
{
	"id": "620dba4ddc874f53b40fe6a6d5fcb412",
	"project": 49208,
	"release": "72.0000000069",
	"platform": "javascript",
	"culprit": "blob:http://app.ft.com/c17928f6-5136-457c-9f06-10497ce065c7 in ?",
	"message": "undefined: TypeError: Attempted to assign to readonly property.",
	"datetime": "2015-12-09T20:35:01.000000Z",
	"time_spent": null,
	"tags": [
		["appName", "ft-app"],
		["level", "error"],
		["logger", "javascript"],
		["sentry:release", "72.0000000069"],
		["browser", "Mobile Safari 8.4"],
		["device", "iPhone"],
		["os", "iOS 8.4"],
		["url", "http://app.ft.com/index_page/uk"],
		["sentry:user", "ip:(redacted)"]
	],
	"errors": [{
		"url": "blob:http://app.ft.com/c17928f6-5136-457c-9f06-10497ce065c7",
		"type": "js_generic_fetch_error",
		"value": "<class 'requests.exceptions.InvalidSchema'>"
	}],
	"extra": {
		"session:duration": 908404,
		"context:log": ""
	},
	"fingerprint": "{{ default }}",
	"sentry.interfaces.Exception": {
		"exc_omitted": null,
		"values": [{
			"stacktrace": {
				"has_system_frames": false,
				"frames": [{
					"abs_path": "blob:http://app.ft.com/c17928f6-5136-457c-9f06-10497ce065c7",
					"in_app": false,
					"lineno": 2,
					"colno": 41776,
					"filename": "blob:http://app.ft.com/c17928f6-5136-457c-9f06-10497ce065c7"
				}],
				"frames_omitted": null
			},
			"type": null,
			"value": "TypeError: Attempted to assign to readonly property.",
			"module": null
		}]
	},
	"sentry.interfaces.Http": {
		"url": "http://app.ft.com/index_page/uk",
		"headers": [
			["User-Agent", "Mozilla/5.0 (iPhone; CPU iPhone OS 8_4 like Mac OS X) AppleWebKit/600.1.4 (KHTML, like Gecko) Mobile/12H143"]
		]
	},
	"sentry.interfaces.User": {
		"ip_address": "(redacted)"
	},
	"version": "7"
}
```

**Comment by dcramer** [comment]
@benvinegar is blob: coming from us/tracekit?

**Comment by benvinegar** [comment]
TraceKit. They added blob support a few months ago, and I merged those changes in [last month](https://github.com/getsentry/raven-js/pull/440) for Raven 2.1.0 (still unreleased).

**Comment by dcramer** [comment]
We need to deal w/ this in hashing then. Likely we need to ignore blob: URLs.

**Comment by mattrobenolt** [comment]
This was fixed with 8d274e14e68286850127c6795425fe810d1624ae
